[Canvas] Move expressions used in Canvas only back into Canvas plugin

cqliu1 opened the pull request:
## Summary

Closes [214026](https://github.com/elastic/kibana/issues/214026).

This deletes the following expression plugins/package and moves the code back into Canvas because they are functions and renderers used exclusively by Canvas:
- `expression_image`
- `expression_error`
- `expression_repeat_image`
- `expression_reveal_image`
- `expression_shape`
- `kbn-expression-utils`

This reduces the number of plugins loaded on Kibana start and will only load these functions when using Canvas.

To test, import these Canvas workpad saved objects to enable Canvas:
[test-workpads.ndjson.zip](https://github.com/user-attachments/files/19766567/test-workpads.ndjson.zip)

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...





<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

cqliu1 committed:
Move reveal_image into Canvas and delete expression_reveal_image plugin

Move reveal_image into canvas and delete expression_reveal_image plugin

Move reveal_image and repeat_image to canvas_plugin_src

Move shape and progress into canvas and delete expression_shape plugin

Delete canvas expression plugins

Remove expression plugins frmo tsconfig.base

Remove imports to external expression plugins

Move metric into canvas and delete expression_metric plugin

Move image into canvas and delete expression_image plugin

Move debug and error renderers in Canvas and delete expression_error plugin

Update plugin list

Update package.json

fix error and debug component imports

Fix progress component import

Restore WORKSPACE.bazel

Updated yarn lock

Update supported renderers

Fix rendered_element story

Fix imports

Register renderers

Fix i18n errors

[CI] Auto-commit changed files from 'node scripts/notice'

[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Fix imports

Extract image, metric, repeat_image, reveal_image strings

Remove plugins from kbn-optimizer/limits.yml

Extract strings for progress and shape

Fix i18n dict

Fix type errors

Fix tests

Update types for repeat_image and reveal_image

Fix circular dep

Fix image types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285856)
* Commit: 0e625f0587e3b17c13b038c76c4b54e57adf2c10

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/285856#0195b4b0-04da-49cc-b054-d6cadccb19d8)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285856","number":285856,"commit":"0e625f0587e3b17c13b038c76c4b54e57adf2c10"}],"number":285856}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/287014)
* Commit: f6519765c627381835fd6437ed128f24113397d9

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/287014#0195cdf3-16a0-462b-a39b-77e94fc65bef)

### History
* :broken_heart: [Build #285856](https://buildkite.com/elastic/kibana-pull-request/builds/285856) failed 0e625f0587e3b17c13b038c76c4b54e57adf2c10

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287014","number":287014,"commit":"f6519765c627381835fd6437ed128f24113397d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285856","number":285856,"commit":"0e625f0587e3b17c13b038c76c4b54e57adf2c10"}],"number":287014}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/289262)
* Commit: 8e64d7697005f7b502e8bf1466b9e132946d164c

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/289262#0195f337-9b9f-4bdd-824a-02dab470a76d)

### History
* :broken_heart: [Build #289101](https://buildkite.com/elastic/kibana-pull-request/builds/289101) failed 790a4cb591f86d84b09bacdbc81cb1f3306c6aa5
* :broken_heart: [Build #287014](https://buildkite.com/elastic/kibana-pull-request/builds/287014) failed f6519765c627381835fd6437ed128f24113397d9
* :broken_heart: [Build #285856](https://buildkite.com/elastic/kibana-pull-request/builds/285856) failed 0e625f0587e3b17c13b038c76c4b54e57adf2c10

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289101","number":289101,"commit":"790a4cb591f86d84b09bacdbc81cb1f3306c6aa5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287014","number":287014,"commit":"f6519765c627381835fd6437ed128f24113397d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285856","number":285856,"commit":"0e625f0587e3b17c13b038c76c4b54e57adf2c10"}],"number":289262}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291005)
* Commit: b93de2d9339464f59d9ef6b8ff7535d9c306e40b

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/291005#01961270-3ece-42c2-a593-e6ae5590ac76)

### History
* :broken_heart: [Build #289262](https://buildkite.com/elastic/kibana-pull-request/builds/289262) failed 8e64d7697005f7b502e8bf1466b9e132946d164c
* :broken_heart: [Build #289101](https://buildkite.com/elastic/kibana-pull-request/builds/289101) failed 790a4cb591f86d84b09bacdbc81cb1f3306c6aa5
* :broken_heart: [Build #287014](https://buildkite.com/elastic/kibana-pull-request/builds/287014) failed f6519765c627381835fd6437ed128f24113397d9
* :broken_heart: [Build #285856](https://buildkite.com/elastic/kibana-pull-request/builds/285856) failed 0e625f0587e3b17c13b038c76c4b54e57adf2c10

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291005","number":291005,"commit":"b93de2d9339464f59d9ef6b8ff7535d9c306e40b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289101","number":289101,"commit":"790a4cb591f86d84b09bacdbc81cb1f3306c6aa5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287014","number":287014,"commit":"f6519765c627381835fd6437ed128f24113397d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285856","number":285856,"commit":"0e625f0587e3b17c13b038c76c4b54e57adf2c10"}],"number":291005}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291256)
* Commit: e86499ee97663b897d07fbf51d0a93dac26a3317

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/291256#01961623-6f15-434f-908b-e86ca909e84e)

### History
* :broken_heart: [Build #291005](https://buildkite.com/elastic/kibana-pull-request/builds/291005) failed b93de2d9339464f59d9ef6b8ff7535d9c306e40b
* :broken_heart: [Build #289262](https://buildkite.com/elastic/kibana-pull-request/builds/289262) failed 8e64d7697005f7b502e8bf1466b9e132946d164c
* :broken_heart: [Build #289101](https://buildkite.com/elastic/kibana-pull-request/builds/289101) failed 790a4cb591f86d84b09bacdbc81cb1f3306c6aa5
* :broken_heart: [Build #287014](https://buildkite.com/elastic/kibana-pull-request/builds/287014) failed f6519765c627381835fd6437ed128f24113397d9
* :broken_heart: [Build #285856](https://buildkite.com/elastic/kibana-pull-request/builds/285856) failed 0e625f0587e3b17c13b038c76c4b54e57adf2c10

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291256","number":291256,"commit":"e86499ee97663b897d07fbf51d0a93dac26a3317"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291005","number":291005,"commit":"b93de2d9339464f59d9ef6b8ff7535d9c306e40b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289101","number":289101,"commit":"790a4cb591f86d84b09bacdbc81cb1f3306c6aa5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287014","number":287014,"commit":"f6519765c627381835fd6437ed128f24113397d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285856","number":285856,"commit":"0e625f0587e3b17c13b038c76c4b54e57adf2c10"}],"number":291256}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291279)
* Commit: 0036ede62dc8da872dd8b8f089e331d6c6071a5e

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/291279#01961647-ded7-458c-aadc-8333d95cf3d4)

### History
* :broken_heart: [Build #291256](https://buildkite.com/elastic/kibana-pull-request/builds/291256) failed e86499ee97663b897d07fbf51d0a93dac26a3317
* :broken_heart: [Build #291005](https://buildkite.com/elastic/kibana-pull-request/builds/291005) failed b93de2d9339464f59d9ef6b8ff7535d9c306e40b
* :broken_heart: [Build #289262](https://buildkite.com/elastic/kibana-pull-request/builds/289262) failed 8e64d7697005f7b502e8bf1466b9e132946d164c
* :broken_heart: [Build #289101](https://buildkite.com/elastic/kibana-pull-request/builds/289101) failed 790a4cb591f86d84b09bacdbc81cb1f3306c6aa5
* :broken_heart: [Build #287014](https://buildkite.com/elastic/kibana-pull-request/builds/287014) failed f6519765c627381835fd6437ed128f24113397d9
* :broken_heart: [Build #285856](https://buildkite.com/elastic/kibana-pull-request/builds/285856) failed 0e625f0587e3b17c13b038c76c4b54e57adf2c10

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291279","number":291279,"commit":"0036ede62dc8da872dd8b8f089e331d6c6071a5e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291256","number":291256,"commit":"e86499ee97663b897d07fbf51d0a93dac26a3317"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291005","number":291005,"commit":"b93de2d9339464f59d9ef6b8ff7535d9c306e40b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289101","number":289101,"commit":"790a4cb591f86d84b09bacdbc81cb1f3306c6aa5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287014","number":287014,"commit":"f6519765c627381835fd6437ed128f24113397d9"}],"number":291279}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292125)
* Commit: 90af49f274628c342c02ee861c7740839e8c0454

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/292125#01962170-f971-42a7-a070-185781d9f3e1)

### History
* :broken_heart: [Build #291279](https://buildkite.com/elastic/kibana-pull-request/builds/291279) failed 0036ede62dc8da872dd8b8f089e331d6c6071a5e
* :broken_heart: [Build #291256](https://buildkite.com/elastic/kibana-pull-request/builds/291256) failed e86499ee97663b897d07fbf51d0a93dac26a3317
* :broken_heart: [Build #291005](https://buildkite.com/elastic/kibana-pull-request/builds/291005) failed b93de2d9339464f59d9ef6b8ff7535d9c306e40b
* :broken_heart: [Build #289262](https://buildkite.com/elastic/kibana-pull-request/builds/289262) failed 8e64d7697005f7b502e8bf1466b9e132946d164c
* :broken_heart: [Build #289101](https://buildkite.com/elastic/kibana-pull-request/builds/289101) failed 790a4cb591f86d84b09bacdbc81cb1f3306c6aa5
* :broken_heart: [Build #287014](https://buildkite.com/elastic/kibana-pull-request/builds/287014) failed f6519765c627381835fd6437ed128f24113397d9

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292125","number":292125,"commit":"90af49f274628c342c02ee861c7740839e8c0454"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291279","number":291279,"commit":"0036ede62dc8da872dd8b8f089e331d6c6071a5e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291256","number":291256,"commit":"e86499ee97663b897d07fbf51d0a93dac26a3317"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291005","number":291005,"commit":"b93de2d9339464f59d9ef6b8ff7535d9c306e40b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289101","number":289101,"commit":"790a4cb591f86d84b09bacdbc81cb1f3306c6aa5"}],"number":292125}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292141)
* Commit: 721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-214993)

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/292141#019621f8-7f4a-4b3c-b7c5-3037a72a3877)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/292141#0196221b-694e-4916-8834-4642bab4bc41)

### History
* :broken_heart: [Build #292125](https://buildkite.com/elastic/kibana-pull-request/builds/292125) failed 90af49f274628c342c02ee861c7740839e8c0454
* :broken_heart: [Build #291279](https://buildkite.com/elastic/kibana-pull-request/builds/291279) failed 0036ede62dc8da872dd8b8f089e331d6c6071a5e
* :broken_heart: [Build #291256](https://buildkite.com/elastic/kibana-pull-request/builds/291256) failed e86499ee97663b897d07fbf51d0a93dac26a3317
* :broken_heart: [Build #291005](https://buildkite.com/elastic/kibana-pull-request/builds/291005) failed b93de2d9339464f59d9ef6b8ff7535d9c306e40b
* :broken_heart: [Build #289262](https://buildkite.com/elastic/kibana-pull-request/builds/289262) failed 8e64d7697005f7b502e8bf1466b9e132946d164c

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292141","number":292141,"commit":"721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292125","number":292125,"commit":"90af49f274628c342c02ee861c7740839e8c0454"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291279","number":291279,"commit":"0036ede62dc8da872dd8b8f089e331d6c6071a5e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291256","number":291256,"commit":"e86499ee97663b897d07fbf51d0a93dac26a3317"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291005","number":291005,"commit":"b93de2d9339464f59d9ef6b8ff7535d9c306e40b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/289262","number":289262,"commit":"8e64d7697005f7b502e8bf1466b9e132946d164c"}],"number":292141}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/kibana-presentation (Team:Presentation)

cqliu1 committed:
Fix image upload

cqliu1 committed:
Delete @kbn/expression-utils package

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293357)
* Commit: 6cb99c4ba1d3841bcf0a3ca16113261d7575aabf

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/293357#01963a38-ff05-479d-baf3-6297f0c1a3d9)

### History
* :broken_heart: [Build #292908](https://buildkite.com/elastic/kibana-pull-request/builds/292908) failed ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd
* :broken_heart: [Build #292141](https://buildkite.com/elastic/kibana-pull-request/builds/292141) failed 721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5
* :broken_heart: [Build #292125](https://buildkite.com/elastic/kibana-pull-request/builds/292125) failed 90af49f274628c342c02ee861c7740839e8c0454
* :broken_heart: [Build #291279](https://buildkite.com/elastic/kibana-pull-request/builds/291279) failed 0036ede62dc8da872dd8b8f089e331d6c6071a5e
* :broken_heart: [Build #291256](https://buildkite.com/elastic/kibana-pull-request/builds/291256) failed e86499ee97663b897d07fbf51d0a93dac26a3317
* :broken_heart: [Build #291005](https://buildkite.com/elastic/kibana-pull-request/builds/291005) failed b93de2d9339464f59d9ef6b8ff7535d9c306e40b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293357","number":293357,"commit":"6cb99c4ba1d3841bcf0a3ca16113261d7575aabf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292908","number":292908,"commit":"ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292141","number":292141,"commit":"721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292125","number":292125,"commit":"90af49f274628c342c02ee861c7740839e8c0454"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291279","number":291279,"commit":"0036ede62dc8da872dd8b8f089e331d6c6071a5e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291256","number":291256,"commit":"e86499ee97663b897d07fbf51d0a93dac26a3317"}],"number":293357}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/dashboards   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export --update'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293371)
* Commit: e9f54f4208bcae72c414b098d54bc70649470f2f

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/293371#01963a49-049a-4d8b-9392-b3426b077181)

### History
* :broken_heart: [Build #293357](https://buildkite.com/elastic/kibana-pull-request/builds/293357) failed 6cb99c4ba1d3841bcf0a3ca16113261d7575aabf
* :broken_heart: [Build #292908](https://buildkite.com/elastic/kibana-pull-request/builds/292908) failed ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd
* :broken_heart: [Build #292141](https://buildkite.com/elastic/kibana-pull-request/builds/292141) failed 721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5
* :broken_heart: [Build #292125](https://buildkite.com/elastic/kibana-pull-request/builds/292125) failed 90af49f274628c342c02ee861c7740839e8c0454
* :broken_heart: [Build #291279](https://buildkite.com/elastic/kibana-pull-request/builds/291279) failed 0036ede62dc8da872dd8b8f089e331d6c6071a5e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293371","number":293371,"commit":"e9f54f4208bcae72c414b098d54bc70649470f2f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293357","number":293357,"commit":"6cb99c4ba1d3841bcf0a3ca16113261d7575aabf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292908","number":292908,"commit":"ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292141","number":292141,"commit":"721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292125","number":292125,"commit":"90af49f274628c342c02ee861c7740839e8c0454"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291279","number":291279,"commit":"0036ede62dc8da872dd8b8f089e331d6c6071a5e"}],"number":293371}
buildkite-pr-comment-->

cqliu1 committed:
Fix image_upload

cqliu1 committed:
Merge branch 'move-expressions-into-canvas' of https://github.com/cqliu1/kibana into move-expressions-into-canvas

cqliu1 committed:
Merge branch 'main' of https://github.com/elastic/kibana into move-expressions-into-canvas

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293491)
* Commit: 84bf695c9686c062d208f36a46278a574885ce0c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-214993)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 1226 | 1311 | +85 |
| `expressionError` | 32 | - | -32 |
| `expressionImage` | 24 | - | -24 |
| `expressionMetric` | 10 | - | -10 |
| `expressionRepeatImage` | 22 | - | -22 |
| `expressionRevealImage` | 25 | - | -25 |
| `expressionShape` | 56 | - | -56 |
| total |  |  | -84 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `@kbn/expression-utils` | 18 | - | -18 |
| `expressionError` | 15 | - | -15 |
| `expressionImage` | 26 | - | -26 |
| `expressionMetric` | 27 | - | -27 |
| `expressionRepeatImage` | 32 | - | -32 |
| `expressionRevealImage` | 14 | - | -14 |
| `expressionShape` | 146 | - | -146 |
| total |  |  | -278 |

#### Any counts in public APIs
> Total count of every any typed public API. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats any` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `@kbn/expression-utils` | 2 | - | -2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 1017.1KB | 1.1MB | +60.9KB |
| `expressionError` | 15.1KB | - | -15.1KB |
| `expressionImage` | 18.9KB | - | -18.9KB |
| `expressionMetric` | 462.0B | - | -462.0B |
| `expressionRepeatImage` | 26.5KB | - | -26.5KB |
| `expressionRevealImage` | 21.8KB | - | -21.8KB |
| total |  |  | -21.9KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `expressionError` | 2 | - | -2 |
| `expressionRevealImage` | 3 | - | -3 |
| total |  |  | -5 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 14.6KB | 13.6KB | -1011.0B |
| `expressionError` | 5.4KB | - | -5.4KB |
| `expressionImage` | 5.7KB | - | -5.7KB |
| `expressionMetric` | 6.0KB | - | -6.0KB |
| `expressionRepeatImage` | 6.4KB | - | -6.4KB |
| `expressionRevealImage` | 6.5KB | - | -6.5KB |
| `expressionShape` | 19.4KB | - | -19.4KB |
| total |  |  | -50.3KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `@kbn/expression-utils` | 18 | - | -18 |
| `expressionError` | 17 | - | -17 |
| `expressionImage` | 26 | - | -26 |
| `expressionMetric` | 32 | - | -32 |
| `expressionRepeatImage` | 32 | - | -32 |
| `expressionRevealImage` | 14 | - | -14 |
| `expressionShape` | 148 | - | -148 |
| total |  |  | -287 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 14 | 24 | +10 |
| `expressionError` | 4 | - | -4 |
| `expressionImage` | 1 | - | -1 |
| `expressionMetric` | 1 | - | -1 |
| `expressionRepeatImage` | 2 | - | -2 |
| `expressionRevealImage` | 2 | - | -2 |
| total |  |  | -0 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 29 | 33 | +4 |
| `expressionError` | 5 | - | -5 |
| `expressionImage` | 1 | - | -1 |
| `expressionMetric` | 1 | - | -1 |
| `expressionRepeatImage` | 1 | - | -1 |
| `expressionRevealImage` | 1 | - | -1 |
| `expressionShape` | 1 | - | -1 |
| total |  |  | -6 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/907cd5904b3328d5c89da31fb19c0528a6acdd50) | [after](https://github.com/elastic/kibana/commit/84bf695c9686c062d208f36a46278a574885ce0c) | diff |
| --- | --- | --- | --- |
| `canvas` | 32 | 36 | +4 |
| `expressionError` | 5 | - | -5 |
| `expressionImage` | 1 | - | -1 |
| `expressionMetric` | 1 | - | -1 |
| `expressionRepeatImage` | 1 | - | -1 |
| `expressionRevealImage` | 1 | - | -1 |
| `expressionShape` | 1 | - | -1 |
| total |  |  | -6 |


</details>


### History
* :broken_heart: [Build #293393](https://buildkite.com/elastic/kibana-pull-request/builds/293393) failed 66ed94d98c8e7b9588c41d3cc04d06be484e525e
* :broken_heart: [Build #293371](https://buildkite.com/elastic/kibana-pull-request/builds/293371) failed e9f54f4208bcae72c414b098d54bc70649470f2f
* :broken_heart: [Build #293357](https://buildkite.com/elastic/kibana-pull-request/builds/293357) failed 6cb99c4ba1d3841bcf0a3ca16113261d7575aabf
* :broken_heart: [Build #292908](https://buildkite.com/elastic/kibana-pull-request/builds/292908) failed ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd
* :broken_heart: [Build #292141](https://buildkite.com/elastic/kibana-pull-request/builds/292141) failed 721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293491","number":293491,"commit":"84bf695c9686c062d208f36a46278a574885ce0c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293393","number":293393,"commit":"66ed94d98c8e7b9588c41d3cc04d06be484e525e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293371","number":293371,"commit":"e9f54f4208bcae72c414b098d54bc70649470f2f"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293357","number":293357,"commit":"6cb99c4ba1d3841bcf0a3ca16113261d7575aabf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292908","number":292908,"commit":"ecfcc41f1d553ddf7e78a2611b0aa601852ff5dd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292141","number":292141,"commit":"721ee1a0d332cfb7458e2ef17b1f3d1e3acf46d5"}],"number":293491}
buildkite-pr-comment-->

nreese reviewed (approved):
kibana-presentation changes LGTM. Great to see page load reduction size and only loading these expressions when canvas app is rendered.

kc13greiner reviewed (approved):
LGTM!

andreadelrio reviewed (approved):
Moved scss files.

TinaHeiligers reviewed (approved):
removing manifest files LGTM

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/14502187946

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 214993
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

cqliu1 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/218597">](https://github.com/elastic/kibana/pull/218597)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

thomasneirynck commented:
This PR shows a nice performance bump in the time it takes to bootstrap Kibana front-end code ("the spinner" on first page load of a user session).

In the benchmarks, this metric, `kibana_loaded` drops around 150ms from 7300 to 7150ms. The drop is consistent and reproducible.

<img width="1070" alt="image" src="https://github.com/user-attachments/assets/58cd4119-19e8-4b61-871c-29fbfe2e679d" />

While the baseline high (+7 seconds), it's been very hard to bring that bootstrap-time down over the years. Seeing a real improvement here is very encouraging. It is a great indication that many packages bloat the overall Kibana footprint, especially when they contain code that is not re-used across plugins. Thx @cqliu1 !
